### DISCUSSION TITLE
Does not implement [LivewireUI\Modal\Contracts\ModalComponent] interface.

### DISCUSSION THREAD

**Discussion by marcwitteveen** [wire-elements/modal#217] [Q&A]
I extended from ModalComponent but still I am getting an error "Exception
[App\Http\Livewire\EditPriceAlertException] does not implement [LivewireUI\Modal\Contracts\ModalComponent] interface."

Anybody can point me in the right direction to solve this?

```
<?php

namespace App\Http\Livewire;

use LivewireUI\Modal\ModalComponent;

class EditPriceAlertException extends ModalComponent
{
    public function render()
    {
        return view('livewire.edit-price-alert-exception');
    }
}
```

**Comment by PhiloNL** [comment]
That's strange, this should work. Have you tried a fresh install of Laravel & Livewire to rule out any influence by your existing codebase or third-party packages by any chance? That would help to debug.

**Comment by safeboot** [comment]
I seem to be having the same issue, I installed it on my project and here's all the things I did:

- `composer require wire-elements/modal`
- `php artisan vendor:publish --tag=livewire-ui-modal-views`
- `php artisan make:livewire Dashboard/Superadmin/CreateUser`

I didn't touch the Livewire controller nor the view (except for inputting 'test' inside the divs) but it shouldn't be giving me an error?

I am running on PHP 8.0 and Laravel 9.8 with TALL stack and Turbolinks.

**Comment by PhiloNL** [comment]
I haven't been able to reproduce this issue. If someone could create a repository that replicates the problem, I can take a further look.

**Reply by holiccoder** [reply]
please try with laravel 11, livewire 3 on laravel herd.

**Comment by safeboot** [comment]
My bad, it seems like the `extends Component` was there by default and I forgot to change it, after adding the safelist into Tailwind, everything seems to be doing alright 👍

**Reply by holiccoder** [reply]
what did you add in the safelist into tailwind?
